projects/schematics/src/shared/utils/file-utils.ts

```diff
@@ -626,6 +705,40 @@ function shouldRemoveImportAndParam(
   return true;
 }
 
+export function removeInjectImports(
+  source: ts.SourceFile,
+  constructorNode: ts.Node,
+  paramToRemove: ClassType
+): Change[] {
+  const importRemovalChange: Change[] = [];
+
+  if (!paramToRemove.injectionToken) {
+    return [new NoopChange()];
+  }
+
+  if (!hasDuplicateDecorator(constructorNode, INJECT_DECORATOR))
+    importRemovalChange.push(
+      removeImport(source, {
+        className: INJECT_DECORATOR,
+        importPath: ANGULAR_CORE,
+      })
+    );
+
+  if (
```

Comment:
You explained me this offline, but it would be nice if we can add a comment here that explains which use case this covers.